Source/Initialization/WarpXInitData.cpp

```diff
@@ -1024,7 +1032,10 @@ WarpX::InitializeExternalFieldsOnGridUsingParser (
                 if((field=='E' and ly(i, j, k)<=0) or (field=='B' and Sy(i, j, k)<=0))  return;
 #elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
                 //In XZ and RZ Ey is associated with a mesh node, so we need to check if  the mesh node is covered
-                if((field=='E' and (lx(i, j, k)<=0 || lx(i-1, j, k)<=0 || lz(i, j, k)<=0 || lz(i, j-1, k)<=0)) or
+                if((field=='E' and (lx(std::min(i  , lx_hi.x-1), std::min(j  , lx_hi.y-1), k)<=0
+                                 || lx(std::max(i-1, lx_lo.x  ), std::min(j  , lx_hi.y-1), k)<=0
+                                 || lz(std::min(i  , lz_hi.x-1), std::min(j  , lz_hi.y-1), k)<=0
+                                 || lz(std::min(i  , lz_hi.x-1), std::max(j-1, lz_lo.y  ), k)<=0)) or
```

Comment:
@WeiqunZhang @RemiLehe 
Not sure I understand why I needed to subtract `1` from `a_hi.<x,z>` (computed above as `a.end`) in order for this to work. 

The out-of-bound error looked like
```
(33,-1,0,0) is out of bound (-1:32,-1:33,0:0,0:0)
```
but `a.end` gives `(33,34,1)` which seems inconsistent with the error message.

Reply:
Answered by @WeiqunZhang: the attributes `begin` and `end` follow the C++ convention. For indexing that follows the Fortran convention (starting from `0`), one could use `amrex::lbound(amrex::Box(a))` or `amrex::ubound(amrex::Box(a))` instead.